Use `is`, not ==, to check exact type matches in _python_dispatch

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #161455
* #161432
* #161329
* #161328
* #161317
* #161315
* #161308
* __->__ #161304
* #161292
* #161301

`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

swolchok committed:
Use `is`, not ==, to check exact type matches in _python_dispatch

`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

[ghstack-poisoned]

swolchok committed:
Update on "Use `is`, not ==, to check exact type matches in _python_dispatch"


`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

[ghstack-poisoned]

Skylion007 reviewed (approved):
We have a flake8/ruff rule for this btw, but it's not enabled.

swolchok committed:
Update on "Use `is`, not ==, to check exact type matches in _python_dispatch"


`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

[ghstack-poisoned]

swolchok committed:
Update on "Use `is`, not ==, to check exact type matches in _python_dispatch"


`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

[ghstack-poisoned]

swolchok committed:
rebase on "Use `is`, not ==, to check exact type matches in _python_dispatch"


`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

[ghstack-poisoned]

swolchok committed:
Update on "Use `is`, not ==, to check exact type matches in _python_dispatch"


`is` checks object identity and is more efficient. Google seems to confirm it is the correct way to do an exact type check.

[ghstack-poisoned]
